File: FE-NextJS/src/model/common/BaseResponse.ts
import { MessageType } from '@/common/enums/MessageEnum';

export interface IErrorList {
  errorCode: string;
  errorMessage: any;
}

export interface IBaseResponse {
  timestamp: string;
  messageType: MessageType;
  messageCode: string;
  message: string;
  errorList: IErrorList[];
  body: any;
}

File: BE/FastAPI/app/controllers/response/base_response.py
from datetime import datetime
from typing import Generic, TypeVar

from pydantic import BaseModel

from app.common.enum.message_type import MessageType
from app.common.utils.string_utils import to_camel_case

T = TypeVar('T')


class BaseResponse(BaseModel, Generic[T]):
    time_stamp: datetime = datetime.now()
    message_code: int | str | None = None
    message_type: MessageType | None = None
    message: str | None = None
    body: T | object | None = None
    error_list: dict | None = None

    class Config:
        alias_generator = to_camel_case
